src/main/java/ai/shreds/infrastructure/InfrastructureEntityMapper.java

```diff
@@ -0,0 +1,124 @@
+package ai.shreds.infrastructure;
+
+import ai.shreds.domain.DomainProductEntity;
+import ai.shreds.domain.DomainPurchaseTransactionEntity;
+import ai.shreds.adapter.AdapterPurchaseRequestDTO;
+import ai.shreds.adapter.AdapterProductDTO;
+import ai.shreds.adapter.AdapterPurchaseDataDTO;
+import org.springframework.stereotype.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class InfrastructureEntityMapper {
+
+    private static final Logger logger = LoggerFactory.getLogger(InfrastructureEntityMapper.class);
+
+    /**
+     * Maps a DomainPurchaseTransactionEntity to an InfrastructurePurchaseTransactionRepositoryImpl.
+     * @param domainEntity the domain entity to be mapped
+     * @return the mapped infrastructure entity
+     */
+    public InfrastructurePurchaseTransactionEntity mapDomainToInfrastructure(DomainPurchaseTransactionEntity domainEntity) {
```

Comment:
Change InfrastructurePurchaseTransactionEntity to DomainPurchaseTransactionEntity